src/Components/MainApp/App.js

```diff
@@ -1,50 +1,48 @@
-import '../../CSS/MainApp/App.css';
-import React, { useState } from 'react';
-import Header from './MainAppHeader.js';
-import ExperimentOne from '../ExperimentOne/ExperimentOne.js';
-import ExperimentTwo from '../ExperimentTwo/ExperimentTwo.js';
-
+import "../../CSS/MainApp/App.css";
+import React, { useState } from "react";
+import Header from "./MainAppHeader.js";
+import ExperimentOne from "../ExperimentOne/ExperimentOne.js";
+import ExperimentTwo from "../ExperimentTwo/ExperimentTwo.js";
 
 function App() {
   // Button States
-  const [headerState, setHeaderState]=useState(true);
-  const [experimentOneButtonState,setExperimentOneButtonState ]=useState(false);
-  const [experimentTwoButtonState,setExperimentTwoButtonState]=useState(false);
-
-  const handleExperimentOneEvent=()=>
-  {
-    setExperimentOneButtonState(true);
-    setExperimentTwoButtonState(false);
-    setHeaderState(false);
-  }
-
-  const handleExperimentTwoEvent=()=>
-  {
-    setExperimentTwoButtonState(true);
-    setExperimentOneButtonState(false);
-    setHeaderState(false);
-  }
-
-  const backButtonState=()=>
-  {
-    setExperimentOneButtonState(false);
-    setExperimentTwoButtonState(false);
-    setHeaderState(true);
-  }
+  const [experimentButtonState, setExperimentButtonState] = useState("0");
+  const experimentButtons = [
+    {
+      name: "Experiment 1",
+      id: "1",
+    },
+    {
+      name: "Experiment 2",
+      id: "2",
+    },
+  ];
 
   return (
+    <div>
       <div>
-        <div>
-          {headerState && <Header/>}
-          {experimentOneButtonState && <ExperimentOne isBackButtonClicked={backButtonState}/>}
-          {experimentTwoButtonState && <ExperimentTwo />}
-        </div>
-
-        <div>
-          {headerState && <button onClick={handleExperimentOneEvent}>Experiment 1</button>}
-          {headerState && <button onClick={handleExperimentTwoEvent}>Experiment 2</button>}
-        </div>
+        {experimentButtonState === "0" && <Header />}
+        {experimentButtonState === "1" && (
+          <ExperimentOne
+            isBackButtonClicked={() => setExperimentButtonState("0")}
+          />
+        )}
+        {experimentButtonState === "2" && <ExperimentTwo />}
+      </div>
+      <div>
+        {experimentButtons.map(
+          (button) =>
+            experimentButtonState === "0" && (
+              <button
+                key={button.id}
```

Comment:
Maybe out of scope for this PR but might want a `ExperimentButton` component which just takes an ID as a prop and will render out these bits using that ID to save some repetition? What do you think?

Reply:
Just to confirm my understanding, you mean, I should another arrow function ExperimentButton in "App.js" and that component basically returns the lines you have highlighted. Although the loop would still exists and would repeat the this logic : ` ExperimentButton(id)`

If so, I will add this as separate issue and then create a new PR for this. As you said to try and maintain PR scope as a good general practice 😄

Reply:
What I'm thinking here is creating a new react component (in a new file) which you could then use here as `<ExperimentButton id={someId} />` replacing the current `<button />` html element that you have. Hope that makes sense!